pallets/democracy-mbm/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+                    let doubled_remaining = remaining_blocks.saturating_mul(2);
+
+                    // 3. Add it to the current block number to get the new end
+                    status.end = current_block_number
+                        .saturating_add(doubled_remaining)
+                        .into();
+                }
+                ReferendumInfo::Finished { .. } => {
+                    // continue;
+                }
+            }
+
+            ReferendumInfoOf::<T>::insert(&last_key, ref_info.clone());
+
+            MigrationState::ReferendumInfo(last_key)
+        } else {
+            MigrationState::StartingVotingOf
+        }
+    }
+
+    fn migrate_voting_of(
+        maybe_last_key: Option<&T::AccountId>,
+        current_block_number: u32,
+    ) -> StepResultOf<T> {
+        let mut iter = if let Some(last_key) = maybe_last_key {
+            VotingOf::<T>::iter_from(VotingOf::<T>::hashed_key_for(last_key))
+        } else {
+            VotingOf::<T>::iter()
+        };
+
+        if let Some((last_key, mut voting)) = iter.next() {
+            match &mut voting {
+                Voting::Direct { prior, .. } => {
+                    let lock_amount = prior.locked();
+
+                    if !lock_amount.is_zero() {
+                        // 1. Calculate the remaining blocks
+                        // as the field block_number is private in PriorLock enum
+                        // it encodes the enum and decodes the 4 bytes (as it's an u32)
+                        let encoded = prior.encode();
+                        let unlock_block_number =
+                            u32::from_le_bytes([encoded[0], encoded[1], encoded[2], encoded[3]]);
+                        let remaining_blocks = unlock_block_number
+                            .saturating_sub(current_block_number)
+                            .saturating_mul(2);
+                        let extended_time =
+                            current_block_number.saturating_add(remaining_blocks).into();
+
+                        // 2. Clean the lock by setting block number and balance to 0
+                        prior.rejig(u32::MAX.into());
+
+                        // 3. Save the lock with migrated values
+                        prior.accumulate(extended_time, lock_amount);
+                    }
+                }
+                Voting::Delegating { prior, .. } => {
+                    let lock_amount = prior.locked();
+
+                    if !lock_amount.is_zero() {
+                        // 1. Calculate the remaining blocks
+                        // as the field block number is private in PriorLock enum
+                        // we encode the enum and decode the 4 bytes (as it's an u32)
+                        let encoded = prior.encode();
+                        let unlock_block_number =
+                            u32::from_le_bytes([encoded[0], encoded[1], encoded[2], encoded[3]]);
```

Comment:
just suggestion, 
```suggestion
                            u32::from_le_bytes((&encoded[..4]).try_into().expect("encoded len < 4; should not happen"));
```

Reply:
Do you want to add a panic in the runtime ? :)
I think this is deterministic enough to be `safe`.

Reply:
Yeah true